doc/source/data/loading-data.rst

```diff
@@ -1,607 +1,531 @@
 .. _loading_data:
 
-====================
+============
 Loading Data
-====================
+============
 
-:class:`Datasets <ray.data.Dataset>` can be created from:
+Ray Data loads data from various sources. This guide shows you how to:
 
-* generated synthetic data,
-* local and distributed in-memory data, and
-* local and external storage systems (local disk, cloud storage, HDFS, etc.).
+* `Read files <#reading-files>`_ like images
+* `Load in-memory data <#loading-in-memory-data>`_ like pandas DataFrames
+* `Read databases <#reading-databases>`_ like MySQL
 
-.. _dataset_generate_data:
+To view the full list of supported sources, see the :ref:`Input/Output reference <input-output>`.
 
--------------------------
-Generating Synthetic Data
--------------------------
+Reading files
+=============
+
+Ray Data reads files from disk or cloud storage.
+
+Reading files from disk
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To read files from disk, call a function like ``ray.data.read_*`` and specify paths with
+the ``local://`` schema. Paths can point to files or directories.
+
+.. tip::
+
+    If your files are accessible on every node, exclude ``local://`` to optimize
+    reads.
 
 .. tab-set::
 
-    .. tab-item:: Int Range
+    .. tab-item:: Images
```

Comment:
Arguably, too many options would hurt readability. I've added TFRecords and Binary as a compromise.

Reply:
Hmm I missed this. I think keeping range() and range_tensor() are pretty important. Almost all users playing around with APIs are going to want to use these as synthetic datasources.

Reply:
FWIW from what I've seen from user code, the "play around with API" is usually creating a dummy pandas dataframe and calling `from_pandas`.

`range` and `range_tensor` are primarily useful for our CI testing or benchmarking. I don't think many users are using this.

Reply:
I would challenge this, since performance testing is a core user activity. There isn't a better way to create large scale test datasets besides using these API calls.

Reply:
ok we can have a "Creating Synthetic Datasets" at the bottom, and clarify this would be useful for generating large datasets for performance testing.

Don't think this should be the first section on the page.